### DISCUSSION TITLE
Environment variables when using ssh

### DISCUSSION THREAD

**Discussion by sla-te** [amoffat/sh#702] [Q&A] [answered]
According to the docs we can run subcommands based of any command (including ssh), how does passing `_env` apply to subcommands? if i do:

```python
import sh
myserver = sh.ssh(...)
new_env = os.environ.copy()
new_env["NEW_VAR"] = "1"
myserver.printenv(_env=new_env)
```

Would the environment variable I set actually get passed through the SSH session and become usable when I call printenv or a command, that would require the variable to be set?

**Comment by amoffat** [comment, marked as answer]
It would be available to the ssh process, and if the ssh process forwards environment variables to the connected server, then yes, but I strongly suspect the ssh process doesn't do that for security reasons.
